### PR TITLE
Add missing TypeScript definitions

### PR DESCRIPTION
**Pull request by matthewjamesadam**
Hello! Thanks for writing spacetime, I've used it in a number of projects, it is great!

Here is a small PR to add some missing things from the TS definitions.  It wasn't clear to me if some of these were omitted intentionally (maybe some of these are deprecated methods?) so let me know if any of these should be removed.

I also added some tests for the methods and constructor syntax while I was at it.

### CODE REVIEW

File: types/types.d.ts [hunk 1]
```diff
@@ -204,6 +225,8 @@ export interface Spacetime {
   monthName(): string
   /** set the current month as a string, like 'april' */
   monthName(value: string): Spacetime
+
+  weekStart(value: number | string): Spacetime
```
Comment (jecraig) [inline types/types.d.ts:229]: weekStart needs documentation

**Review by jecraig — APPROVED** [review]

File: types/types.d.ts (hunk 1 above)
Comment (matthewjamesadam) [reply types/types.d.ts:230]: Hah, serves me right for not checking my PR before opening.  Thanks, fixed.

**Review by jecraig — APPROVED** [review]

**Comment by spencermountain** [comment]
thanks Matt! Clever idea.

**Comment by spencermountain** [comment]
released as `6.11.0`

**Pull request merged**